coderd/jwtutils/jwt.go

```diff
@@ -0,0 +1,28 @@
+package jwtutils
+
+import (
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+)
+
+const (
+	keyIDHeaderKey = "kid"
+)
+
+// Claims defines the payload for a JWT. Most callers
+// should embed jwt.Claims
+type Claims interface {
+	Validate(jwt.Expected) error
+}
+
+// ParseOptions are options for parsing a JWT.
+type ParseOptions struct {
```

Comment:
TBH these could be two separate types to make it harder to use incorrectly. I don't think any real code benefits from these being the same except for the test code.